src/vm/methodtable.cpp

```diff
@@ -2915,13 +2957,13 @@ bool MethodTable::ClassifyEightBytesWithNativeLayout(SystemVStructRegisterPassin
             }
         }
 
-        if ((normalizedFieldOffset % fieldNativeSize) != 0)
+        if ((normalizedFieldOffset % pFieldMarshaler->AlignmentRequirement()) != 0)
```

Comment:
Why was this change needed?

Reply:
Fixed arrays have the alignment requirement of their type instead of the size of the array. For example, an `int[3]` can be aligned on a 4 byte boundary. It doesn't have to be on a 12 byte boundary. The same situation happens with structures and classes.